src/array/cpu/spmm_blocking_libxsmm.h

```diff
@@ -80,10 +80,16 @@ inline void SpMMCreateBlocks(
   if (num_K_blocks > 1) {
     IdType *indptr_block_buf = reinterpret_cast<IdType *>(aligned_alloc(
         64, (M_block_size + 1) * num_M_blocks * num_K_blocks * sizeof(IdType)));
-    IdType *indices_block_buf = reinterpret_cast<IdType *>(
-        aligned_alloc(64, indptr[M] * sizeof(IdType)));
-    IdType *edges_block_buf = reinterpret_cast<IdType *>(
-        aligned_alloc(64, indptr[M] * sizeof(IdType)));
+    IdType *indices_block_buf = nullptr;
+    if (use_lhs) {
+      indices_block_buf = reinterpret_cast<IdType *>(
+          aligned_alloc(64, indptr[M] * sizeof(IdType)));
+    }
+    IdType *edges_block_buf = nullptr;
+    if (use_rhs) {
+      edges_block_buf = reinterpret_cast<IdType *>(
+          aligned_alloc(64, indptr[M] * sizeof(IdType)));
+    }
```

Comment:
Could you add a comment on why this change fixes the memory leak?  As far as I remember `aligned_alloc(3)` doesn't require using `free(3)` to reclaim memory (https://linux.die.net/man/3/aligned_alloc).  Since I didn't see any `free(3)` call I think there should be other reasons.

Reply:
Memory, which is  allocated here:
https://github.com/dmlc/dgl/pull/5210/files#diff-09f913927175aed9edd405df276b75d0c56b07ca917a887ab5ed19344e960f07R83-R92
![image](https://user-images.githubusercontent.com/71560847/216276257-9ff7307b-299f-459f-82f2-fe7677499d1e.png)

gets deallocated here:
https://github.com/dmlc/dgl/pull/5210/files#diff-09f913927175aed9edd405df276b75d0c56b07ca917a887ab5ed19344e960f07R401-R411
![image](https://user-images.githubusercontent.com/71560847/216276513-b5513c5b-88ae-40b0-8613-a483999a7f2f.png)

As you can see memory is deallocated only when "use_lhs" or "use_rhs" flags are equal to "true". So to avoid a leak, appropriate "if statements" have to be added before allocation.

aligned_alloc requires using free in order to avoid a memory leak according to the spec https://en.cppreference.com/w/c/memory/aligned_alloc